winch/codegen/src/isa/x64/masm.rs

```diff
@@ -2543,6 +2543,183 @@ impl Masm for MacroAssembler {
         Ok(())
     }
 
+    fn v128_trunc_sat(
+        &mut self,
+        context: &mut CodeGenContext<Emission>,
+        kind: V128TruncSatKind,
+    ) -> Result<()> {
+        self.ensure_has_avx()?;
+
+        let reg = writable!(context.pop_to_reg(self, None)?.reg);
+        let scratch = writable!(regs::scratch_xmm());
+
+        match kind {
```

Comment:
Each arm here is considerably long, could we extract each into it's own helper? I had a bit of a hard time following along, having each arm in its own function will make it easier (at least personally) to reason about each invariant.